bin/v-add-mail-domain-webmail

```diff
@@ -128,46 +128,46 @@ else
 		fi
 	fi
 
-	add_webmail_config "$WEB_SYSTEM" "${WEBMAIL_TEMPLATE}.tpl"
+	add_webmail_config ${WEB_SYSTEM} ${WEBMAIL_TEMPLATE}.tpl
 
 	if [ -n "$PROXY_SYSTEM" ]; then
-		add_webmail_config "$PROXY_SYSTEM" "${PROXY_TEMPLATE}.tpl"
+		add_webmail_config ${PROXY_SYSTEM} ${PROXY_TEMPLATE}.tpl
 	fi
 
 	# Enable SSL for webmail if available
-	if [ -f $HOMEDIR/$user/conf/mail/$domain/ssl/$domain.crt ] || [ "$SSL" = 'yes' ]; then
-		add_webmail_config "$WEB_SYSTEM" "${WEBMAIL_TEMPLATE}.stpl"
+	if [ -f ${HOMEDIR}/${user}/conf/mail/${domain}/ssl/${domain}.crt ] || [ "$SSL" = 'yes' ]; then
+		add_webmail_config ${WEB_SYSTEM} ${WEBMAIL_TEMPLATE}.stpl
 
 		if [ -n "$PROXY_SYSTEM" ]; then
-			add_webmail_config "$PROXY_SYSTEM" "${PROXY_TEMPLATE}.stpl"
+			add_webmail_config ${PROXY_SYSTEM} ${PROXY_TEMPLATE}.stpl
 		fi
 	fi
 fi
 
-WEBMAIL=$(get_object_value 'web' 'DOMAIN' "$domain" "$WEBMAIL")
+WEBMAIL=$(get_object_value 'web' 'DOMAIN' ${domain} ${WEBMAIL})
 if [ -z "$WEBMAIL" ]; then
-	add_object_key 'mail' 'DOMAIN' "$domain" 'WEBMAIL' 'SSL'
+	add_object_key 'mail' 'DOMAIN' ${domain} 'WEBMAIL' 'SSL'
 fi
 
 # Set SSL as enabled in configuration
-update_object_value 'mail' 'DOMAIN' "$domain" '$WEBMAIL' "$webmail"
+update_object_value 'mail' 'DOMAIN' ${domain} '$WEBMAIL' ${webmail}
 
 #----------------------------------------------------------#
 #                       Hestia                             #
 #----------------------------------------------------------#
 
 if [ "$restart" = 'yes' ]; then
 	# Restarting web server
-	$BIN/v-restart-web "$restart"
+	${BIN}/v-restart-web "$restart"
 	check_result $? "Web restart failed" > /dev/null
 
-	$BIN/v-restart-proxy "$restart"
+	${BIN}/v-restart-proxy "$restart"
 	check_result $? "Proxy restart failed" > /dev/null
 fi
 
 # Logging
 if [ "$quiet" != 'yes' ]; then
-	$BIN/v-log-action "$user" "Info" "Mail" "Webmail access enabled (Domain: $domain)."
+	${BIN}/v-log-action ${user} "Info" "Mail" "Webmail access enabled (Domain: ${domain})."
```

Comment:
Same here...

Reply:
I know the ${variable} method as better practice than "$variable" method to avoid unsafe usage of $variable (without quotes). Yes, it will be used also for arrays, groups or default definitions of strings. But if you use ${variable}, you do not need more "${variable}". Spaces or other special characters can be used and does not lead to "brocken value", which you can avoid otherwise only using quotes "$variable".

For example, this OSS project uses this method:
https://github.com/Freetz-NG/freetz-ng/blob/master/fwmod

I use this trick with this definition of ${variable} often, if the variable is a part of path. In this case I define path as path_variable=/main_folder/${subfolder1}/${subfolder2} instead of path_variable="/mainfolder/$subfolder1/$subfolder2".

From my point of view this method with ${var} is better than "$var", if one use it consequently.

In HESTIA the construct like $BIN/start_script is used, which is very dangerous also from security point of view, since, if you define $BIN using space, you can execute a fully other code. Therefore you should use "$BIN/start_script" always, or ${BIN}/start_script, as proposed from me. I find personally the method with curly braces better.

Reply:
Enough talking, test it yourself...
```
cat << "EOF" > ./test.sh
#!/bin/bash

[ -n "$1" ] && echo "param 1 = $1"
[ -n "$2" ] && echo "param 2 = $2"
[ -n "$3" ] && echo "param 3 = $3"
EOF

chmod +x ./test.sh

./test.sh 1 "$null" 3

./test.sh 1 ${null} 3
```

Reply:
maybe you are right with this special case with `$user` (special usage as call argument), but I can construct for you here also a very dangerous example with using of $BIN in the line 170 without braces to execute an alien code or alien script. Therefore in generally using of curved spaces is better, than this wild scripting with `"$var"` or without `$var`

Reply:
Please use {curly braces} only where necessary, not abusing everywhere...

Enclosing variables in double quotes is standard safe coding practice.

Reply:
I will correct it on the positions, if it is used as argument. That make sense for me. But all other cases is my personal programming style and with the links above I argumented, that I am not alone with such a programming style: Curved braces is an alternative to achive the same effect, as using `"$var"` method, which is in most cases heaver readable in my point of view.  Therefore I will use it, equal, if you please it or not. And this method is for me also a "standard safe coding practice" as `"$var"` for you, excepted such special cases, like these empty arguments during call a script.

Reply:
That's not "my standard"

I won't list the CVEs related to {curly braces} here...

Perhaps the following example can show more clearly:
```
cat << "EOF" > ./test.sh
#!/bin/bash

${1}
"$1"
EOF

chmod +x ./test.sh

./test.sh "echo evil"
```

Reply:
First: I do not use these `${1}` or `${2}` or something directly for call, but as `aaa=${1}`
Second: HESTIA uses actually in all binaries such constructes like `$HESTIA/path/script` `$BIN/path/script` (without quotation marks), which are also potentially vurnelable. Is your proposal to change all these lines by `"$BIN"/path/script` ?

Reply:
Correct!

And just to clarify, I'm not accusing you.:handshake:

Almost all current Hestia CLIs have potential variable vulnerability...

Honestly, it's not our responsibility.

But it would be nice to add protection by the way as we make changes.

Given {curly braces}'s bad history, they really should be avoided.

Even when necessary, be sure to enclose it in double quotes `"${var:=some}"`

Reply:
The other motivation of use the curly braces is the combination with `"..."`, if you have such constructs like `"first"second"more"` using quotation marks inside of quotation marks. But it must be checked separatelly, too. I do not have now an exact example for it, but I had already cases, in which I could not solve it with quotation marks only. Maybe variable inside of sed or something similar. Ok, I will check it double with curly braces.

Reply:
You mean?
```
sed 's/old-string/'"$new"'-string/g' # works and safely
sed "s/old-string/$new-string/g"     # works and safely
sed "s/old-string/${new}_string/g"   # works and safely, necessary use cases for {curly braces}
```